apps/web/src/services/administration/iam/user/modules/user-management-modal/modules/NotificationEmailForm.vue

```diff
@@ -0,0 +1,333 @@
+    isEdit: false,
+    isSent: false,
+    isCollapsed: true,
+    loginUserId: computed(() => store.state.user.userId),
+});
+const formState = reactive({
+    email: '' || props.email,
+    verificationCode: '',
+});
+const validationState = reactive({
+    isEmailValid: undefined as undefined | boolean,
+    emailInvalidText: '' as TranslateResult | string,
+    isValidationCodeValid: undefined as undefined | boolean,
+    validationCodeInvalidText: '' as TranslateResult | string,
+});
+
+/* Components */
+const inputValidation = () => {
+    if (formState.email === '') {
+        validationState.isEmailValid = false;
+        return;
+    } if (emailValidator(formState.email)) {
+        validationState.isEmailValid = false;
+        validationState.emailInvalidText = i18n.t('IDENTITY.USER.FORM.EMAIL_INVALID');
+        return;
+    }
+    validationState.isEmailValid = true;
+    validationState.emailInvalidText = '';
+};
+const handleChangeInput = () => {
+    inputValidation();
+    emit('change-input', { ...formState, email: formState.email });
+};
+const handleClickChange = () => {
+    if (!state.isEdit) {
+        state.isEdit = true;
+    }
+};
+const setForm = () => {
+    formState.email = props.item.email || '';
+};
+const handleEditButton = () => {
+    state.isSent = false;
+    state.isEdit = true;
+};
+
+/* API */
+const handleClickSend = async () => {
+    state.loading = true;
+    try {
+        await postValidationEmail({
+            user_id: props.item.user_id,
+            email: formState.email,
+            domain_id: props.item.domain_id,
+        });
+        emit('change-verify', false);
+
+        if (state.loginUserId === props.item.user_id) {
+            await store.dispatch('user/setUser', { email: formState.email });
+        }
+        state.isSent = true;
+    } catch (e) {
+        ErrorHandler.handleError(e);
+    } finally {
+        state.loading = false;
+    }
+};
+const handleChangeVerify = async () => {
+    state.loading = true;
+    try {
+        await postValidationCode({
+            user_id: props.item.user_id,
+            domain_id: props.item.domain_id,
+            code: formState.verificationCode,
+        }, state.loginUserId === props.item.user_id);
+        if (userPageState.visibleUpdateModal) {
+            state.isSent = false;
+            state.isEdit = false;
+            state.isCollapsed = true;
+            emit('change-verify', true);
+        }
+    } catch (e) {
+        validationState.isValidationCodeValid = true;
+        validationState.validationCodeInvalidText = vm.$t('COMMON.NOTIFICATION_MODAL.INVALID_CODE');
+    } finally {
+        state.loading = false;
+    }
+};
+
+/* Init */
+(async () => {
+    if (userPageState.visibleUpdateModal) {
+        await setForm();
+    }
+})();
+
+watch(() => props.email, () => {
+    formState.email = props.email;
+});
+watch(() => props.isValidEmail, () => {
+    state.isEdit = !props.isValidEmail;
+}, { immediate: true });
```

Comment:
nitpick: Please use callback value in watch().

```suggestion
watch(() => props.email, (value) => {
    formState.email = value;
});
watch(() => props.isValidEmail, (value) => {
    state.isEdit = !value;
}, { immediate: true });
```